crates/burn-jit/src/kernel/pool/max_pool2d_backward.rs

```diff
@@ -1,322 +1,81 @@
-
-        cpa!(scope, oh_start_tmp = signed_ih + signed_padding_0);
-        cpa!(scope, oh_start_tmp = oh_start_tmp - kms_0);
-        cpa!(scope, oh_start_tmp = oh_start_tmp / signed_pool_stride_0);
-
-        cpa!(scope, ow_start_tmp = signed_iw + signed_padding_1);
-        cpa!(scope, ow_start_tmp = ow_start_tmp - kms_1);
-        cpa!(scope, ow_start_tmp = ow_start_tmp / signed_pool_stride_1);
-
-        cpa!(scope, oh_start_tmp = max(oh_start_tmp, 0i32));
-        cpa!(scope, ow_start_tmp = max(ow_start_tmp, 0i32));
-
-        let oh_start = scope.create_local(u32::as_elem());
-        let ow_start = scope.create_local(u32::as_elem());
-
-        cpa!(scope, oh_start = cast(oh_start_tmp));
-        cpa!(scope, ow_start = cast(ow_start_tmp));
-
-        let oh_end_tmp = scope.create_local(Elem::Int(IntKind::I32));
-        let ow_end_tmp = scope.create_local(Elem::Int(IntKind::I32));
-
-        cpa!(scope, oh_end_tmp = max(kms_0, 0i32));
-        cpa!(scope, ow_end_tmp = max(kms_1, 0i32));
-
-        let oh_end = scope.create_local(u32::as_elem());
-        let ow_end = scope.create_local(u32::as_elem());
-
-        let oh_end_limit = scope.create_local(u32::as_elem());
-        let ow_end_limit = scope.create_local(u32::as_elem());
-
-        cpa!(scope, oh_end = cast(oh_end_tmp));
-        cpa!(scope, ow_end = cast(ow_end_tmp));
-
-        cpa!(scope, oh_end = oh_end + oh_start);
-        cpa!(scope, oh_end_limit = grad_shape_2 - 1u32);
-
-        cpa!(scope, ow_end = ow_end + ow_start);
-        cpa!(scope, ow_end_limit = grad_shape_3 - 1u32);
-
-        cpa!(scope, oh_end = min(oh_end, oh_end_limit));
-        cpa!(scope, ow_end = min(ow_end, ow_end_limit));
-
-        let index_current = scope.create_local(u32::as_elem());
-        let index_current_tmp = scope.create_local(u32::as_elem());
-
-        cpa!(scope, index_current = ih * output_stride_2);
-        cpa!(scope, index_current_tmp = iw * output_stride_3);
-        cpa!(scope, index_current += index_current_tmp);
-
-        cpa!(scope, oh_end = oh_end + 1u32);
-        cpa!(scope, ow_end = ow_end + 1u32);
-
-        (oh_start, oh_end, ow_start, ow_end)
-    }
-}
-
-impl<R: JitRuntime, E: JitElement> Kernel for MaxPool2dWithIndicesBackwardEagerKernel<R, E> {
-    fn define(&self) -> KernelDefinition {
-        let mut scope = Scope::root();
-        let item = E::cube_elem().into();
-
-        let indices = Variable::new(
-            VariableKind::GlobalInputArray(0),
-            Item::new(Elem::Int(IntKind::I32)),
-        );
-        let grad = Variable::new(VariableKind::GlobalInputArray(1), item);
-        let output = Variable::new(VariableKind::GlobalOutputArray(0), item);
-
-        scope.write_global_custom(output);
-
-        MaxPool2dBackwardComputeShader {
-            indices,
-            grad,
-            output,
-            kernel_size: self.kernel_size,
+            if index_max == index_current {
```

Comment:
Can we use select?
`grad_acc += select(index_max == index_current, grad[index], 0)`